### DISCUSSION TITLE
volume integral of a variable over specified blocks

### DISCUSSION THREAD

**Discussion by mcacace** [idaholab/moose#16138] [Q&A General] [answered]
Dear all,
I would like to compute the volume integral of a certain value over a block of my input mesh. 
My idea is:
a) compute the (volume integral) value of the variable over each element
b) sum (a) over all elements of the block of interest
Now, I know that (a) could be done relatively easily within existing PostProcessor (inheriting from elementIntegralValues). However, I don't seem to find any existing ones for (b) already implemented in the framework. I noticed that there is a SumNodalVar post processor, but I do not think it works properly for my specific case. 
Was wondering if anyone could help in this regard. 
Thanks a lot in advance,
mauro

**Comment by aeslaughter** [comment, marked as answer]
I think ElementAverageValue is what you need. You can use the 'block' parameter to limit it to subdomain
https://mooseframework.inl.gov/source/postprocessors/ElementAverageValue.html

**Reply by mcacace** [reply]
ups, I missed it... It does read as the pp I was looking for. One question though: Does the average is done on all elements belonging to a block? In other words: value = sum(value_integral_over_element)/total_block_volume?
Thanks,
mauro

**Reply by aeslaughter** [reply]
It computes the average forth the block volume, when the 'block' parameter is used.
